lib/web/apiserver.go

```diff
@@ -5162,15 +5164,105 @@ func (h *Handler) authExportPublic(w http.ResponseWriter, r *http.Request, p htt
 	)
 	if err != nil {
 		h.log.WithError(err).Debug("Failed to generate CA Certs.")
-		http.Error(w, err.Error(), trace.ErrorToCode(err))
+		h.serveErrorResponse(err, w, r)
 		return
 	}
 
-	reader := strings.NewReader(authorities)
+	err = serveTextResponse(authorities, w, r)
+	if err != nil {
+		h.log.WithError(err).Error("Failed to encode and send response.")
+		h.serveErrorResponse(err, w, r)
+	}
+}
+
+// Given an HTTP request and a response body to the request, return an encoded response based
+// on the `Accept` header, as well as the appropriate Content-Type.
+// If an error has occured, the response encoding and content type are chosen on a best-effort basis.
+// Will always return encoded response bytes, and content type value.
+func encodeTextResponse(responseBody string, r *http.Request) ([]byte, string, error) {
+	// Track the latest error. Callers can choose to handle or ignore this. Useful for
+	// error handling functions.
+	var trackedErr error
+
+	// Parse "Accept" header values into mime types
+	var mimeTypes []string
+	// Handle the case of multiple headers
+	for _, acceptHeaderValue := range r.Header["Accept"] {
+		// Handle the case of multiple values in a single header
+		for _, splitHeaderValue := range strings.Split(acceptHeaderValue, ",") {
+			splitHeaderValue = strings.TrimSpace(splitHeaderValue)
+
+			// Parameters (such as charset and quality factor) are not currently supported
+			var mimeType string
+			mimeType, _, err := mime.ParseMediaType(splitHeaderValue)
+			if err != nil {
+				trackedErr = trace.Wrap(err, "malformed mime type %q", splitHeaderValue)
+				continue
+			}
+
+			mimeTypes = append(mimeTypes, mimeType)
+		}
+	}
+
+	// Ignore duplicate MIME types
+	mimeTypes = slices.Compact(mimeTypes)
+
+	// Add a default value in the event that either no value is sent, or no supported value is sent
+	// This must always be a plain text response to preserve backwards compatibility
+	mimeTypes = append(mimeTypes, "text/plain")
```

Comment:
💅 I think we can swap these two assignments.
Otherwise, we can end up with two `text/plain` (which is not a big deal, anyway).